packages/runtime-core/src/hydrationStrategies.ts

```diff
@@ -86,21 +86,24 @@ export const hydrateOnInteraction: HydrationStrategyFactory<
   }
 
 export function forEachElement(node: Node, cb: (el: Element) => void): void {
-  // fragment
-  if (isComment(node) && node.data === '[') {
-    let depth = 1
-    let next = node.nextSibling
-    while (next) {
-      if (next.nodeType === DOMNodeTypes.ELEMENT) {
-        cb(next as Element)
-      } else if (isComment(next)) {
-        if (next.data === ']') {
-          if (--depth === 0) break
-        } else if (next.data === '[') {
-          depth++
+  // #11952
+  if (isComment(node)) {
```

Comment:
TextNode also needs to be filtered out. see [Playground with this PR](https://deploy-preview-11954--vue-sfc-playground.netlify.app/#__SSR__eNqFUk1PAjEQ/SuTXoCEwEFPm4VEjQeNEaPGUy/r7iwUy3TTdvmQ7H932gUkinjYZDvvTfvevNmKq6oaLGsUiUhdblXlwaGvK9AZTUdSeCfFWJJaVMZ62ILFEvpQYKkIr9yG8hvDECH5Psw2hc08TuhNOfWu8ahyV2iEBkprFtDh5zqSJOWGnIeH7HMzIYTRyVu7W0kA2mQF2gS6PRiNoRXT7QyGu95goNPrB+buxeSXmG5PUsOfpHTYGmVbfPC4qDQT+QSQ7sUkbmZWbL/MtEMphoymwyOq6PNkWH+ppoO5M8Tji0KlyFm50mgnlVfsT4oEIhKwTGuzuo81b2uMgmPPDPOPE/W5W4eaFE8WHdolKzlgPrNT9C18+/KIa/4/gAtT1JrZZ8BndEbXQWNLu66JM7JHvKj2Ls5a0fTV3a49ktubCkIDs4l8KTiCENpf1r/lXgwuYx/HwVM8SvCfFWy348mayqXx2hBRAu/GaMw4yWYcIj4TLsRV+xlk8wWwswY8)
maybe we should change the following code
https://github.com/vuejs/core/blob/928af5fe2f5f366b5c28b8549c3728735c8d8318/packages/runtime-core/src/hydrationStrategies.ts#L40
to
```ts
forEach(el => (el instanceof Element) && ob.observe(el))
```